src/extension-runners/firefox-desktop.js

```diff
@@ -0,0 +1,229 @@
+/* @flow */
+
+// Import flow types from npm dependencies.
+import type FirefoxProfile from 'firefox-profile';
+
+import {
+  MultipleExtensionsReloadError,
+  RemoteTempInstallNotSupported,
+  WebExtError,
+} from '../errors';
+import * as defaultFirefoxApp from '../firefox';
+import {
+  connectWithMaxRetries as defaultFirefoxConnector,
+} from '../firefox/remote';
+import {createLogger} from '../util/logger';
+// Import flow types from project files.
+import type {
+  FirefoxRDPResponseAddon,
+  RemoteFirefox,
+} from '../firefox/remote';
+import type {
+  ExtensionRunnerParams,
+} from './base';
+import type {
+  FirefoxPreferences,
+} from '../firefox/preferences';
+import type {
+  FirefoxInfo, // eslint-disable-line import/named
```

Comment:
we already have the same issue elsewhere (in one of the test files if I recall correctly).

I also don't get "the why" yet, while working on the refactoring I decided to defer further investigation, but I'm going to take a look again (as soon as I've applied the rest of the changes needed based on this round of review comments)